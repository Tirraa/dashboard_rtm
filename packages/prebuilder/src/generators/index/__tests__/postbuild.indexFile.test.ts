// eslint-disable-next-line import/no-extraneous-dependencies
import { afterAll, describe, expect, it } from 'vitest';

import generateIndexFile from '../indexFile';

const fs = require('fs/promises');

const __TARGET_FOLDER_ROOT = './packages/prebuilder/src/generators/index/__tests__/FAKE_CODEGEN';
const __TARGET_FOLDER = __TARGET_FOLDER_ROOT + '/' + 'INDEX_FILE';

describe('generateIndexFile', () => {
  afterAll(async () => {
    await fs.rm(__TARGET_FOLDER, { recursive: true });
  });

  it('should match snapshot', async () => {
    await generateIndexFile(__TARGET_FOLDER);

    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/index.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});
