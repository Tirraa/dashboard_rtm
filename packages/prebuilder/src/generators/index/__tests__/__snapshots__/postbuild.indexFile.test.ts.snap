// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateIndexFile > should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
import type Blog from "./Blog";
import type BlogArchitecture from "./BlogArchitecture";
import blogCategories from "./blogCategories";
import type DefaultLanguageToken from "./DefaultLanguageToken";
import type LandingPages from "./LandingPages";
import type Pages from "./Pages";
import pagesTitles from "./pagesTitles";
import type { PagesFromCodegenSchema } from "./Pages";

export { Blog, BlogArchitecture, blogCategories, DefaultLanguageToken, LandingPages, Pages, pagesTitles, PagesFromCodegenSchema };
"
`;

exports[`generateIndexFile should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
import type Blog from "./Blog";
import type BlogArchitecture from "./BlogArchitecture";
import blogCategories from "./blogCategories";
import type DefaultLanguageToken from "./DefaultLanguageToken";
import type LandingPages from "./LandingPages";
import type Pages from "./Pages";
import pagesTitles from "./pagesTitles";
import type { PagesFromCodegenSchema } from "./Pages";

export { Blog, BlogArchitecture, blogCategories, DefaultLanguageToken, LandingPages, Pages, pagesTitles, PagesFromCodegenSchema };
"
`;
